File: tests/fixtures/rsa_purpose/signing.go
// RSA used only for signatures. Must be recommended a signature scheme.
package main

import "crypto/rsa"

func sign(k *rsa.PrivateKey, digest []byte) ([]byte, error) {
	return rsa.SignPSS(nil, k, 0, digest, nil)
}

func verify(k *rsa.PublicKey, digest, sig []byte) error {
	return rsa.VerifyPKCS1v15(k, 0, digest, sig)
}

File: demo/estate/svc-payments/signing.py
"""Payment authorisation signing.

Demonstration 1: RSA used for SIGNING. The PSS padding object is what settles
the purpose, so this must be recommended a signature scheme (ML-DSA) and never
a KEM. Getting this wrong was the defect Milestone 2 existed to fix.
"""

from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.primitives.asymmetric import padding


def sign_authorisation(private_key, payload: bytes) -> bytes:
    return private_key.sign(
        payload,
        padding.PSS(mgf=padding.MGF1(hashes.SHA256()), salt_length=32),
        hashes.SHA256(),
    )
